lib/Conversion/LTLToCore/LTLToCore.cpp

```diff
@@ -0,0 +1,276 @@
+// Conversion patterns
+//===----------------------------------------------------------------------===//
+
+namespace {
+
+// Custom pattern matchers
+
+// Matches and records a boolean attribute
+struct I1ValueMatcher {
+  Value *what;
+  I1ValueMatcher(Value *what) : what(what) {}
+  bool match(Value op) const {
+    if (!op.getType().isSignlessInteger(1))
+      return false;
+    *what = op;
+    return true;
+  }
+};
+
+static inline I1ValueMatcher mBool(Value *const val) {
+  return I1ValueMatcher(val);
+}
+
+// Matches and records an arbitrary op
+template <typename OpType, typename... OperandMatchers>
+struct BindingRecursivePatternMatcher
+    : mlir::detail::RecursivePatternMatcher<OpType, OperandMatchers...> {
+
+  using BaseMatcher =
+      mlir::detail::RecursivePatternMatcher<OpType, OperandMatchers...>;
+  BindingRecursivePatternMatcher(OpType *bop, OperandMatchers... matchers)
+      : BaseMatcher(matchers...), opBind(bop) {}
+
+  bool match(Operation *op) {
+    if (BaseMatcher::match(op)) {
+      *opBind = llvm::cast<OpType>(op);
+      return true;
+    }
+    return false;
+  }
+
+  OpType *opBind;
+};
+
+template <typename OpType, typename... Matchers>
+static inline auto mOpWithBind(OpType *op, Matchers... matchers) {
+  return BindingRecursivePatternMatcher<OpType, Matchers...>(op, matchers...);
+}
+
+struct HasBeenResetOpConversion : OpConversionPattern<verif::HasBeenResetOp> {
+  using OpConversionPattern<verif::HasBeenResetOp>::OpConversionPattern;
+
+  // HasBeenReset generates a 1 bit register that is set to one once the reset
+  // has been raised and lowered at at least once.
+  LogicalResult
+  matchAndRewrite(verif::HasBeenResetOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto i1 = rewriter.getI1Type();
+    // Generate the constant used to set the register value
+    Value constZero = rewriter.create<hw::ConstantOp>(op.getLoc(), i1, 0);
+
+    // Generate the constant used to enegate the
+    Value constOne = rewriter.create<hw::ConstantOp>(op.getLoc(), i1, 1);
+
+    // Create a backedge for the register to be used in the OrOp
+    circt::BackedgeBuilder bb(rewriter, op.getLoc());
+    circt::Backedge reg = bb.get(rewriter.getI1Type());
+
+    // Generate an or between the reset and the register's value to store
+    // whether or not the reset has been active at least once
+    Value orReset =
+        rewriter.create<comb::OrOp>(op.getLoc(), adaptor.getReset(), reg);
+
+    // This register should not be reset, so we give it dummy reset and resetval
+    // operands to fit the build signature
+    Value reset, resetval;
+
+    // Finally generate the register to set the backedge
+    reg.setValue(rewriter.create<seq::CompRegOp>(
+        op.getLoc(), orReset,
+        rewriter.createOrFold<seq::ToClockOp>(op.getLoc(), adaptor.getClock()),
+        reset, resetval, llvm::StringRef("hbr"), constZero));
```

Comment:
You might be able to use the default builder instead, which accepts a parameter for every operand and attribute:
```
  // SeqOps.td: CompRegOp
  let arguments = (ins
    AnyType:$input,
    ClockType:$clk,
    OptionalAttr<StrAttr>:$name,
    Optional<I1>:$reset,
    Optional<AnyType>:$resetValue,
    Optional<AnyType>:$powerOnValue,
    OptionalAttr<InnerSymAttr>:$inner_sym
  );
```
Something like the following could work:
```suggestion
    // Finally generate the register to set the backedge
    reg.setValue(rewriter.create<seq::CompRegOp>(
      op.getLoc(),
      orReset,
      rewriter.createOrFold<seq::ToClockOp>(op.getLoc(), adaptor.getClock()),
      StringAttr{}, // name (could also be rewriter.getStringAttr("hbr"))
      reset,
      resetval,
      constZero,
      InnerSymAttr{} // inner_sym
    ));
```